### DISCUSSION TITLE
CSS variables and functions for real-time customization

### DISCUSSION THREAD

**Discussion by 2acode** [twbs/bootstrap#37767] [Q&A] [answered]
Hi,
I love the idea of being able to take advantage of real-time global customization as assumed in v6:
https://github.com/twbs/bootstrap/discussions/37057

In the meantime I would like to customize the primary button in real time as described here:
https://getbootstrap.com/docs/5.3/components/buttons/#variables

For the 'hover' and 'active' states the 'shade-color' function is used but I can't use it in the css file so how can I set these colors to keep the same BS logic?
At the moment the only solution I've found is to recreate the 'shade-color' function server-side and 'print' the variable in the css.

- Are there any other solutions to customize the primary button real-time (no sass)?
- Where is the 'shade-color' function defined (I couldn't find it)?

Thank you

**Comment by mdo** [comment, marked as answer]
Yes, this is only available in Sass, our pre-processor.

> Are there any other solutions to customize the primary button real-time (no sass)?

Overriding the CSS variables manually. 

> Where is the 'shade-color' function defined (I couldn't find it)?

Inside `_functions.scss`.
